writer.go

```diff
@@ -114,27 +114,42 @@ func recordID(Key sdk.Data) string {
 	return string(key)
 }
 
-func recordPayload(payload sdk.Change) ([]float32, error) {
+type recordPayload struct {
+	Values       []float32    `json:"values"`
+	SparseValues sparseValues `json:"sparse_values"`
+}
+
+func (r recordPayload) PineconeSparseValues() *pinecone.SparseValues {
+	v := &pinecone.SparseValues{r.SparseValues.Indices, r.SparseValues.Values}
+	return v
+}
+
+type sparseValues struct {
+	Indices []uint32  `json:"indices"`
+	Values  []float32 `json:"values"`
+}
+
+func parseRecordPayload(payload sdk.Change) (parsed recordPayload, err error) {
 	data := payload.After
 
 	if data == nil || len(data.Bytes()) == 0 {
-		return nil, errors.New("empty payload")
+		return parsed, errors.New("empty payload")
 	}
 
-	var parsedPayload []float32
-
-	err := json.Unmarshal(data.Bytes(), &parsedPayload)
+	err = json.Unmarshal(data.Bytes(), &parsed)
 	if err != nil {
-		return nil, fmt.Errorf("error unmarshalling JSON: %v", err)
+		return parsed, fmt.Errorf("error unmarshalling JSON: %v", err)
 	}
 
-	return parsedPayload, nil
+	return parsed, nil
 }
 
 func recordMetadata(data sdk.Metadata) (*pinecone.Metadata, error) {
 	convertedMap := make(map[string]interface{})
 	for key, value := range data {
-		convertedMap[key] = value
+		if trimmed, hasPrefix := trimPineconeKey(key); hasPrefix {
```

Comment:
According to https://docs.pinecone.io/guides/data/filtering-with-metadata

pinecone metadata is used to allow custom filtering. My reasoning to exclude conduit specific metadata was that the connector user won't care about grouping vectors by which opencdc version were they written, from which plugin where they written, etc. The readAt / createdAt conduit metadata fields might be useful for grouping, idk.

It felt to me a bit dirty to merge conduit metadata / pinecone metadata in a single pinecone vector.

Reply:
@hariso but now that you mention it, it could be useful for the connector user to have a way to distinguish vectors written by conduit.

If I remove the filtering it keeps the connector simple, so I'm up for removing this logic.

Reply:
Yeah, we can do that for now. A user can always remove metadata in a processor. Or, if we see many requests by users to remove the metadata, we'll change the connector.:)

Reply:
If you ask me, we should stay away from assumptions. Even though this might seem user-friendly and something that the user might want to do anyway, I'd argue that Conduit gets harder to use in general, if I have to check out every connector and its specific user-friendly logic. My expectation is that if a record contains 3 fields of metadata, I want to see them land in Pinecone. If I don't want that metadata, I have processors to tweak the record before it reaches the destination.

Reply:
thanks for the input @lovromazgon !